### DISCUSSION TITLE
hypercable tracker

### DISCUSSION THREAD

**Discussion by hooopo** [HyperCable/hypercable#5] [General]
## basic info

* referer
* user-agent
* url
* page
* event name  (pageview, custom name)
* visit token (short expires)
* visitor token (long expires)
* custom attributes (json)
* utc time

## user
* user id

## multi site
* set site id

## native app
* native app support
* compatible with web
* app_version os_version platform

**Comment by hooopo** [comment]
requirements: https://github.com/HyperCable/hypercable/discussions/4

**Reply by izayl** [reply]
url 和 page 的区别是什么

**Reply by hooopo** [reply]
额，可能需要重新整理一下

最近调研了一下google analytics的tracker，一个新的想法：

使用GA现有的tracker，和扩展功能：https://developers.google.com/analytics/devguides/collection/analyticsjs/tasks

GA允许重放或者替换payload，如果直接利用GA的tracker，对于集成现有使用GA的网站就非常方便。GA的Measurement Protocol文档和参数说明已经很稳定了：

* https://ga-dev-tools.appspot.com/hit-builder/
* https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
* https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide

缺点是：

* mobile的支持（这部分还没仔细看）
* event相关接口的灵活性
* 对复杂payload的支持

所以我的想法是，先利用GA的tracker，直接用。
如果之后有需求，再自己实现tracker，在field语义和名称上可以兼容MP协议，然后格式上其实可以有区别，比如支持json和自定义属性的增强。

**Comment by hooopo** [comment]
这里是重放GA请求到自定义endpoint的插件：

https://github.com/snowplow-incubator/snowplow-google-analytics-plugin
